### PR TITLE
CONVENTIONS.md: add PromQL query to calculate adequate memory request

### PR DESCRIPTION
**Pull request by fpetkovski**
This commit adds a Prometheus query which can be used to calculate the difference between the requested memory of each container, and the actual memory which should be requested according to memory request convention.

### CODE REVIEW

File: CONVENTIONS.md [hunk 1]
```diff
@@ -306,6 +306,18 @@ the Prometheus instance in the cluster to query the
 `kube_pod_resource_request` and `kube_pod_resource_limit` metrics and
```
Comment (simonpasquier) [inline CONVENTIONS.md:306]: should we remove `kube_pod_resource_limit` since it is stated before that cluster components shouldn't have resource limits?

**Review by simonpasquier — COMMENTED** [review]
It might be handy to include the PromQL query to compare CPU requests with actual usage.

**Comment by simonpasquier** [comment]
/assign @dhellmann 

Assigning Doug since he updated this section recently.

File: CONVENTIONS.md (hunk 1 above)
Comment (dhellmann) [reply CONVENTIONS.md:306]: We could do that in another PR, if we're sure nothing actually does set limits.

**Review by dhellmann — COMMENTED** [review]
This looks good to me. I'm not a PromQL expert, so I'll leave it to @simonpasquier to sign off on the details. It would be cool if we could show some sample output, but if that's a graph it probably wouldn't be easy to do here so it's not a big deal.

Thanks!

/approve

File: CONVENTIONS.md [hunk 2]
```diff
@@ -286,6 +286,18 @@ end-to-end job run and the component being tuned uses 350m on the same
 run, the request for the component being tuned should be set to
 100m/600m * 350m ~= 58m.
 
+The following PromQL query illustrates how to calculate the adequate CPU request
+for all containers in the `openshift-monitoring` namespace:
```
Comment (simonpasquier) [inline CONVENTIONS.md:290]: the query mixes the openshift-etcd and openshift-monitoring namespaces.

Comment (fpetkovski) [reply CONVENTIONS.md:290]: The formula for calculating the adequate request depends on the etcd usage and request. This is why we have both the target namespace (openshift-monitoring in this case) and the reference namespace (openshift-etcd). 

Does this resolve the question, or did you have something else in mind>

Comment (simonpasquier) [reply CONVENTIONS.md:290]: Ah right I should have read the paragraph above indeed :)

Having said that all monitoring components (except CMO which runs on control plane nodes) should use SDN as the baseline.

Comment (fpetkovski) [reply CONVENTIONS.md:290]: Ack. The query should be updated to use SDN as the baseline. I also added a note that pods which run on control-plane nods should be evaluated against etcd.

File: CONVENTIONS.md [hunk 3]
```diff
@@ -286,6 +286,22 @@ end-to-end job run and the component being tuned uses 350m on the same
...
+
+```PromQL
+# CPU request / usage of the SDN container
+scalar(
+  max(kube_pod_container_resource_requests{namespace="openshift-sdn", container="sdn", resource="cpu"}) /
+  max(max_over_time(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate{namespace="openshift-etcd", container="etcd"}[120m]))) *
```
Comment (simonpasquier) [inline CONVENTIONS.md:296]:
```suggestion
  max(max_over_time(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate{namespace="openshift-sdn", container="sdn"}[120m]))) * 
```

**Pull request merged**